jdaviz/configs/cubeviz/plugins/spectral_extraction/tests/test_spectral_extraction.py

```diff
@@ -231,3 +231,51 @@ def test_aperture_markers(cubeviz_helper, spectrum1d_cube):
         extract_plg._obj.vue_adopt_slice_as_reference()
         extract_plg._obj.vue_goto_reference_wavelength()
         assert slice_plg.slice == 1
+
+
+def test_cone_aperture(cubeviz_helper, spectrum1d_cube_larger):
+    from specutils import Spectrum1D
+    cubeviz_helper.load_data(spectrum1d_cube_larger)
+    cubeviz_helper.load_regions([CirclePixelRegion(PixCoord(1, 1), radius=0.5)])
+
+    mask_cube = Spectrum1D(flux=np.ones_like(spectrum1d_cube_larger.flux),
+                           spectral_axis=spectrum1d_cube_larger.spectral_axis,
+                           mask=np.ones_like(spectrum1d_cube_larger.flux))
+    cubeviz_helper.load_data(mask_cube, override_cube_limit=True)
+    cubeviz_helper._loaded_mask_cube = cubeviz_helper.app.data_collection[-1]
+
+    extract_plg = cubeviz_helper.plugins['Spectral Extraction']
+    slice_plg = cubeviz_helper.plugins['Slice']
+
+    extract_plg.aperture = 'Subset 1'
+    extract_plg.wavelength_dependent = True
+    assert cubeviz_helper._loaded_mask_cube.get_object(cls=Spectrum1D, statistic=None)
+
+    slice_plg.slice = 1
+    extract_plg._obj.vue_adopt_slice_as_reference()
+    cone_aperture = extract_plg._obj.cone_aperture()
+    ref_wavelength_1 = extract_plg._obj.reference_wavelength
+    assert (cone_aperture.shape ==
+            cubeviz_helper._loaded_flux_cube.get_object(cls=Spectrum1D, statistic=None).shape)
+
+    slice_plg.slice = 8
+    extract_plg._obj.vue_adopt_slice_as_reference()
+    ref_wavelength_2 = extract_plg._obj.reference_wavelength
+    # Not enough slices for the cone to develop
+    cone_aperture_2 = extract_plg._obj.cone_aperture()
+
+    cone_result = [[[1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
```

Comment:
Is this result correct? Looks like your flux is increasing w.r.t. array flattened index and you are at slice index 8, so the "flux" should be pretty high by now. Why is the expected values all ones and zeroes?

https://github.com/spacetelescope/jdaviz/blob/6e08521445cd6b92635c27513193bd8bf26fb369/jdaviz/conftest.py#L207

I cannot quite tell what you are checking here.

Reply:
`cone_aperture` is a method that returns a cube of boolean values.